packages/admin-ui/src/filters/filter-multiple/filter-option-checkbox.tsx

```diff
@@ -1,16 +1,66 @@
-import React from 'react'
+import type { ComponentPropsWithoutRef, Ref } from 'react'
+import React, { forwardRef } from 'react'
 
 import { ComboboxItem } from 'ariakit/combobox'
 import { Checkbox } from '../../checkbox'
 
-import * as style from '../filter.style'
 import { createComponent, useElement } from '@vtex/admin-ui-react'
 import { usePopoverContext } from '../filter-popover-context'
 import type { ComboboxMultipleState } from '../../combobox'
 import type { FilterOption } from '../filter/filter.state'
-import { csx } from '@vtex/admin-ui-core'
+import {
+  filterControlInputTheme,
+  filterControlOptionTheme,
+} from '../filter.css'
+import { cx } from '@vtex/admin-ui-core'
 
-export const FilterOptionCheckbox = createComponent<
+export const FilterOptionCheckbox = forwardRef(function FilterOptionCheckbox(
+  props: FilterOptionCheckboxProps,
+  ref: Ref<HTMLDivElement>
+) {
+  const {
+    id,
+    value,
+    label,
+    className = '',
+    onClick,
+    children,
+    ...htmlProps
+  } = props
+
+  const { state } = usePopoverContext()
+
+  const item = { id, label, value }
+
+  const combobox = state.combobox as ComboboxMultipleState<FilterOption<any>>
+
+  return (
+    <ComboboxItem
+      ref={ref}
+      className={cx(filterControlInputTheme, className)}
+      onClick={(e) => {
+        combobox.onChange(item)
+        onClick?.(e)
+      }}
+      id={id}
+      aria-selected={combobox.isSelected(item)}
+      {...htmlProps}
+    >
+      {children || (
```

Comment:
Do we make sure somewhere that this isn't an empty object (truthy value)?

Reply:
I don't think we need to care about this case because it's the react children, so if there is an empty object is because the user did it, so it's an expected behavior